### ISSUE TITLE
kubernetes service certificate missing SAN for kubernetes.default on hypershift cluster

### ISSUE THREAD

**Issue by ncolon** [openshift/hypershift#1950]
The Kube API endpoint is missing a SAN entry for `kubernetes.default` on a hypershift cluster.

RHACM 2.6 was deployed on a 4.12-rc4 cluster on AWS.

On the RHACM hub cluster, this is what the certificate SAN list looks like 

```
/ # echo |openssl s_client -showcerts -connect $KUBERNETES_SERVICE_HOST:443|openssl x509 -text|grep DNS
 
                DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local,
DNS:openshift, DNS:openshift.default, DNS:openshift.default.svc, DNS:openshift.default.svc.cluster.local, DNS:172.30.0.1,
IP Address:172.30.0.1
```

This is what it looks like on the hypershift cluster
 
```
$ echo |openssl s_client -showcerts -connect $KUBERNETES_SERVICE_HOST:443|openssl x509 -text|grep DNS

                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local,
DNS:kube-apiserver, DNS:kube-apiserver.clusters-hypershift-c1.svc, DNS:kube-apiserver.clusters-hypershift-c1.svc.cluster.local,
DNS:af3337183987b4a1099f2abc592c4095-8000527f677046ae.elb.us-east-2.amazonaws.com,
DNS:api.hypershift-c1.hypershift.local, IP Address:127.0.0.1, IP Address:172.31.0.1
```

Is this by design?

**Comment by nastacio** [comment]
@sjenning, do you know the minor versions of OpenShift where this would have been merged?
- 4.10.x
- 4.11.x
?

**Comment by nastacio** [comment]
FWIW, I just tested a combination of HyperShift host on OCP 4.12.0, creating a HyperShift cluster using OCP 4.10.47, and the issue persists:

```
KUBERNETES_SERVICE_HOST=a7020e8b0ec5141dd96d296cdf67fe44-d34bec9c6b561f22.elb.us-east-1.amazonaws.com:6443

echo |openssl s_client -showcerts -connect $KUBERNETES_SERVICE_HOST |openssl x509 -text|grep kubernetes.default
```

Output

```txt
depth=0 O = kubernetes, CN = kubernetes
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 O = kubernetes, CN = kubernetes
verify error:num=21:unable to verify the first certificate
verify return:1
depth=0 O = kubernetes, CN = kubernetes
verify return:1
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:kube-apiserver, DNS:kube-apiserver.clusters-cloudpak-fixed.svc, DNS:kube-apiserver.clusters-cloudpak-fixed.svc.cluster.local, DNS:a7020e8b0ec5141dd96d296cdf67fe44-d34bec9c6b561f22.elb.us-east-1.amazonaws.com, DNS:api.cloudpak-fixed.hypershift.local, IP Address:127.0.0.1, IP Address:172.31.0.1
```

About to try creating a HyperShift deployment running OCP 4.12.0.

**Comment by nastacio** [comment]
I see that the fix is available on MultiClusterEngine 2.2 (part of RHACM 2.7)
